frontend/src/helpers/API.js

```diff
@@ -1,85 +1,65 @@
+/* eslint-disable no-alert */
+/* eslint-disable no-console */
 import axios from 'axios';
 
-axios.defaults.baseURL = 'http://localhost:3001/api/v3/';
-const currentSessionToken = sessionStorage.getItem('token');
-
-const validateToken = async () => {
-  let validToken = false;
-
-  await axios
-    .post('token/get_user', null, {
-      params: {
-        token: currentSessionToken,
-      },
-    })
-    .then((response) => {
-      const [status, data] = [response.data.status, response.data];
-
-      if (response.data.status === 200) {
-        validToken = true;
-      }
-
-      if (status === 400 && data.error === 'Invalid') {
-        sessionStorage.clear('token');
-      }
-
-      // TODO: HANDLE SESSION TIMEOUT
-    });
-  return validToken;
-};
+axios.defaults.baseURL = 'http://localhost:3001/api/v3';
 
-const signIn = async (login, password, setLoginError) => {
-  let signInSuccessful = false;
-  await axios
-    .post('token/create', null, {
-      params: {
-        login,
-        password,
-      },
-    })
-    .then((response) => {
-      const [status, data] = [response.status, response.data];
+// Axios instance we can import and use in our apis
+const axiosInstance = axios.create({
+  baseURL: 'http://localhost:3001/api/v3',
+});
 
-      if (status === 200 && data.token) {
-        setLoginError();
-        sessionStorage.setItem('token', data.token);
-        signInSuccessful = true;
-        window.location.reload();
-      }
-
-      if (status === 401) {
-        setLoginError(response.data.error);
-      }
-    });
-  return signInSuccessful;
-};
-
-const signOut = (setLoginOutError) => {
-  axios
-    .post('token/delete', null, {
-      params: {
-        token: currentSessionToken,
-      },
-    })
-    .then((response) => {
-      const [status, data] = [response.status, response.data];
+// For validation we need to send the token with every request to the backend
+const currentSessionToken = sessionStorage.getItem('token');
 
-      if (status === 200 || (status === 401 && data.error === 'Invalid')) {
-        sessionStorage.clear('token');
-        setLoginOutError();
-        window.location.reload();
-      }
+/* We intercept the request to ensure the session token is sent in the params
+   on every route except login */
+axiosInstance.interceptors.request.use((config) => {
+  const newConfig = config;
+  if (window.location.pathname !== '/login') {
+    newConfig.params = {
+      token: currentSessionToken,
+      ...newConfig.params,
+    };
+  }
+  return newConfig;
+});
 
-      if (status === 401 && data.error !== 'Invalid') {
-        setLoginOutError(data.error);
+// TODO: REPLACE ALL ALERTS WITH PROPER HANDLING
+/* Use interceptors to handle error responses generalized by status code
+   We can customize the response using the pathname */
+axiosInstance.interceptors.response.use(
+  (response) => response,
+  // eslint-disable-next-line consistent-return
+  (error) => {
```

Comment:
this catch does not appear to work.  I suggest something like as below
NOTES: 
- I just put in an alert (we need to write the code to handle the actual error
- If the system hits an error here it still needs to pass something back. I suggest simply passing back "false"
- I also included "suggested usage" section (based on what I started for object_types)

SUGGESTED CODE:
  (error) => {
    const error_response = error.response;
    switch (error_response["status"]) {
      case 400:
        alert("400: "+JSON.stringify(error_response["data"]));
        break;
      case 401:
        if (window.location.pathname !== '/login') {
          if (currentSessionToken) {
            axios.post('/token/delete');
            sessionStorage.clear('token');
          }
        }
        /* TODO: HANDLE SESSION TIMEOUT
           if (...pathname !== '/login' && message !== 'Session timeout') { OPEN LOGIN MODAL} */
        alert("401: "+JSON.stringify(error_response["data"]));
        break;
      case 403:
        // TODO: HANDLE PERMISSIONS
        alert("403: "+JSON.stringify(error_response["data"]));
        break;
      case 404:
        alert("404: "+JSON.stringify(error_response["data"]));
        break;
      default:
        alert(error_response["status"]+": "+JSON.stringify(error_response["data"]));
    }
    // Return something obvious to stop processing. I like simply "false"
    return false;
  },

SUGGESTED USAGE 
   ...
    // make an API call    
    await objectsAPI.create(formData);

    // break if the HTTP call resulted in an error (the "return false" above)
    // NOTE: the alert("break") is just there for testing. Whatever processing should be handled in API.js, and we just need stop the system from trying to continue...
    if (!response) {
      alert("break") 
      return;
    }

    // process errors (this is defined by our API architecture, we <either> return a success <or> { "errors" => ... }
    // NOTE: in this example I am processing by putting up an AlertToast
    const errors = response["errors"];
    if (errors) {
      return setAlertProps({
        message: JSON.stringify(errors, null, 2),
        severity: 'error',
        open: true,
      });
    }

    // success
    ... continue processing